cookiecutter/main.py

```diff
@@ -52,7 +52,8 @@ def expand_abbreviations(template, config_dict):
     return template
 
 
-def cookiecutter(template, checkout=None, no_input=False, extra_context=None):
+def cookiecutter(template, checkout=None, no_input=False,
```

Comment:
I am afraid this breaks backwards compatibility :worried: 

We should append the new kwarg at the very end.